tests/testthat/test-conditional_formatting.R

```diff
@@ -1,259 +1,394 @@
-  expect_equal(exp, wb$worksheets[[11]]$conditionalFormatting)
+  exp <- c(`A1:A11` = '<cfRule type="dataBar" priority="1"><dataBar showValue="1">\n                          <cfvo type="min"/><cfvo type="max"/>\n                          <color rgb="FF638EC6"/>\n                          </dataBar>\n                          <extLst><ext uri="{B025F937-C7B1-47D3-B67F-A62EFF666E3E}" xmlns:x14="http://schemas.microsoft.com/office/spreadsheetml/2009/9/main"><x14:id>{F7189283-14F7-4DE0-9601-54DE9DB40000}</x14:id></ext>\n                        </extLst></cfRule>')
+  expect_identical(exp, wb$worksheets[[1]]$conditionalFormatting)
+  expect_save(wb)
+})
 
+test_that("type = 'between' works", {
+  wb <- my_workbook()
   wb$add_worksheet("between")
   ## Between
   # Highlight cells in interval [-2, 2]
   wb$add_data("between", -5:5)
-  wb_conditional_formatting(wb, "between", cols = 1, rows = 1:11, type = "between", rule = c(-2, 2))
+  wb$add_conditional_formatting(
+    "between",
+    cols = 1,
+    rows = 1:11,
+    type = "between",
+    rule = c(-2, 2)
+  )
 
-  exp <- c(`A1:A11` = "<cfRule type=\"cellIs\" dxfId=\"0\" priority=\"1\" operator=\"between\"><formula>-2</formula><formula>2</formula></cfRule>")
-  expect_equal(exp, wb$worksheets[[12]]$conditionalFormatting)
+  exp <- c(`A1:A11` = '<cfRule type="cellIs" dxfId="0" priority="1" operator="between"><formula>-2</formula><formula>2</formula></cfRule>')
+  expect_identical(exp, wb$worksheets[[1]]$conditionalFormatting)
+  expect_save(wb)
+})
 
+test_that("type = 'topN' works", {
+  wb <- my_workbook()
   wb$add_worksheet("topN")
   ## Top N
   wb$add_data("topN", data.frame(x = 1:10, y = rnorm(10)))
   # Highlight top 5 values in column x
-  wb_conditional_formatting(wb, "topN", cols = 1, rows = 2:11,
-                        style = posStyle, type = "topN", rank = 5)#'
+  wb$add_conditional_formatting(
+    "topN",
+    cols = 1,
+    rows = 2:11,
+    style = posStyle,
+    type = "topN",
+    params = list(rank = 5)
+  )
+
   # Highlight top 20 percentage in column y
-  wb_conditional_formatting(wb, "topN", cols = 2, rows = 2:11,
-                        style = posStyle, type = "topN", rank = 20, percent = TRUE)
+  wb$add_conditional_formatting(
+    "topN",
+    cols = 2,
+    rows = 2:11,
+    style = posStyle,
+    type = "topN",
+    params = list(rank = 20, percent = TRUE)
+  )
 
   exp <- c(
-    `A2:A11` = "<cfRule type=\"top10\" dxfId=\"1\" priority=\"2\" rank=\"5\" percent=\"NULL\"></cfRule>",
-    `B2:B11` = "<cfRule type=\"top10\" dxfId=\"1\" priority=\"1\" rank=\"20\" percent=\"1\"></cfRule>"
+    `A2:A11` = '<cfRule type="top10" dxfId="1" priority="2" rank="5" percent="NULL"></cfRule>',
```

Comment:
oh it is `NULL` here

Reply:
Works fine with `"0"` and is easier to handle